crates/oxc_isolated_declarations/src/global_symbol_binding_tracker.rs

```diff
@@ -0,0 +1,176 @@
+    computed_properties_using_non_global_symbol: FxHashSet<Span>,
+    computed_properties_using_non_global_global_this: FxHashSet<Span>,
+}
+
+impl<'a> GlobalSymbolBindingTracker<'a> {
+    pub fn new(allocator: &'a Allocator) -> Self {
+        let ast = AstBuilder::new(allocator);
+        let levels = ast.vec1(Scope::new());
+        Self {
+            levels,
+            computed_properties_using_non_global_symbol: FxHashSet::default(),
+            computed_properties_using_non_global_global_this: FxHashSet::default(),
+        }
+    }
+
+    fn does_computed_property_reference_non_global_symbol(&self, key: &PropertyKey) -> bool {
+        self.computed_properties_using_non_global_symbol.contains(&key.span())
+    }
+
+    fn does_computed_property_reference_non_global_global_this(&self, key: &PropertyKey) -> bool {
+        self.computed_properties_using_non_global_global_this.contains(&key.span())
+    }
+
+    pub fn does_computed_property_reference_well_known_symbol(&self, key: &PropertyKey) -> bool {
+        if let PropertyKey::StaticMemberExpression(expr) = key {
+            if let Expression::Identifier(identifier) = &expr.object {
+                identifier.name == "Symbol"
+                    && !self.does_computed_property_reference_non_global_symbol(key)
+            } else if let Expression::StaticMemberExpression(static_member) = &expr.object {
+                if let Expression::Identifier(identifier) = &static_member.object {
+                    identifier.name == "globalThis"
+                        && static_member.property.name == "Symbol"
+                        && !self.does_computed_property_reference_non_global_global_this(key)
+                } else {
+                    false
+                }
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    }
+
+    fn has_in_scope_symbol_binding(&self) -> bool {
+        self.levels.iter().any(|level| level.has_symbol_binding)
+    }
+
+    fn has_in_scope_global_this_binding(&self) -> bool {
+        self.levels.iter().any(|level| level.has_global_this_binding)
+    }
+
+    fn handle_name_binding(&mut self, name: &Atom) {
+        match name.as_str() {
+            "Symbol" => {
+                let scope = self.levels.last_mut().unwrap();
+                scope.has_symbol_binding = true;
+            }
+            "globalThis" => {
+                let scope = self.levels.last_mut().unwrap();
+                scope.has_global_this_binding = true;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl<'a> Visit<'a> for GlobalSymbolBindingTracker<'a> {
+    fn enter_scope(&mut self, _: ScopeFlags, _: &Cell<Option<ScopeId>>) {
+        let scope = Scope::new();
+        self.levels.push(scope);
+    }
+
+    fn leave_scope(&mut self) {
+        self.levels.pop();
+    }
+
+    fn visit_ts_type(&mut self, _: &TSType<'a>) {
+        // Optimization: we don't need to traverse down into types.
```

Comment:
There might be other paths that can be avoided, but I'm not sure which other ones would provide as much value.